rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/DatabaseSanitation/Sw360DatabaseSanitationService.java

```diff
@@ -0,0 +1,96 @@
+
+import static org.eclipse.sw360.datahandler.common.CommonUtils.allAreEmptyOrNull;
+import static org.eclipse.sw360.datahandler.common.CommonUtils.oneIsNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.THttpClient;
+import org.apache.thrift.transport.TTransportException;
+import org.eclipse.sw360.datahandler.thrift.RequestStatus;
+import org.eclipse.sw360.datahandler.thrift.SW360Exception;
+import org.eclipse.sw360.datahandler.thrift.components.ComponentService;
+import org.eclipse.sw360.datahandler.thrift.licenses.LicenseService;
+import org.eclipse.sw360.datahandler.thrift.projects.ProjectService;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class Sw360DatabaseSanitationService {
+
+    @Value("${sw360.thrift-server-url:http://localhost:8080}")
+    private String thriftServerUrl;
+    
+    public Map<String, Object> dublicateIdentifiers(User sw360User, HttpServletRequest request) throws TException {
+        Map<String, Object> responseMap = new HashMap<>();
+
+        Map<String, List<String>> duplicateComponents=null;
+        Map<String, List<String>> duplicateReleases=null;
+        Map<String, List<String>> duplicateReleaseSources=null;
+        Map<String, List<String>> duplicateProjects=null;
+        try {
+            ComponentService.Iface componentClient = getThriftComponentClient();
+            ProjectService.Iface projectClient = getThriftProjectClient();
+            duplicateComponents = componentClient.getDuplicateComponents();
+            duplicateReleases = componentClient.getDuplicateReleases();
+            duplicateReleaseSources = componentClient.getDuplicateReleaseSources();
+            duplicateProjects =  projectClient.getDuplicateProjects();
+        } catch (SW360Exception exp) {
+            if (exp.getErrorCode() == 404) {
+                throw new ResourceNotFoundException(exp.getWhy());
+            } else {
+                throw new RuntimeException(exp.getWhy());
+            }
+        }
+        if (oneIsNull(duplicateComponents, duplicateReleases, duplicateProjects, duplicateReleaseSources)) {
+            throw new HttpMessageNotReadableException("Some duplicates are null");
```

Comment:
`HttpMessageNotReadableException`  is used for client side error. Seems like this is not applicable here and same in line `75`

Reply:
done.